hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/txn/UserTxnScope.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.workflows.handle.flow.txn;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import javax.inject.Scope;
+
+/**
+ * Scope for bindings whose lifetime consists of a single platform transaction.
+ * This scope contains the following bindings:
+ * <ol>
+ *     <li>HederaState</li>
+ *     <li>configuration</li>
+ *     <li>TokenContext</li>
+ *     <li>RecordListBuilder</li>
+ *     <li>PreHandleResult</li>
+ *     <li></li>
+ *  </ol>
+ */
```

Comment:
I'm confused how this scope is different than `@UserDispatchScope`?

Reply:
UserTxnScope includes anything that happens only once for platform transaction. Like staking period hook, genesis record hook. Since we consider user and child as dispatches now to re-use the code for both, `UserDispatchScope` includes all objects that are shared in `DispatchProcessr` and `HandleLogic` only for User transaction.

Reply:
👍 Not necessary in this PR, but we may want to point that out in docs somewhere